documentation/src/main/asciidoc/reference/introduction.adoc

```diff
@@ -1053,10 +1053,25 @@ Hibernate Reactive session associated with the context.
 === Vert.x instance service
 
 The `VertxInstance` service defines how Hibernate Reactive obtains an instance
-of Vert.x. The default implementation just creates one the first time it's
-needed. But if your program requires control over how the Vert.x instance is
-created, or how it's obtained, you can override the default implementation and
-provide your own `VertxInstance`. Let's consider this example:
+of Vert.x when there is no instance associated with the calling thread. The
```

Comment:
> when there is no instance associated with the calling thread

Technically, it also defines how Hibernate Reactive obtains Vert.x even if there is  one associated to the Thread.
Isn't this a bit misleading?

Reply:
Ummmmm. Hrrrrmmmm. Yeah, right. See there's a bit of a slightly confusing interplay between `VertxContext` and `VertxInstance` here. TBH I've never completely understood the significance of:

```java
public void execute(Runnable runnable) {
    io.vertx.core.Context context = vertxInstance.getVertx().getOrCreateContext();
    if ( Vertx.currentContext() == context ) {
        runnable.run();
    }
    else {
        context.runOnContext( x -> runnable.run() );
    }
}
```

Like .... why don't we just use the `Vertx.currentContext()` if there is one? Why do we insist on it being the one coming from "our" Vert.x?

I mean, I know why we do it that way: because that's what @cescoffier and @vietj told us to do, but I think I need to understand it better.

Reply:
1. When I talk about "an instance of Vert.x", I mean `io.vertx.core.Vertx`.
2. When I talk about "a Vert.x context", I mean `io.vertx.core.Context`.

So, `DefaultVertxInstance` is responsible for `1`, `org.hibernate.reactive.context.impl.VertxContext` is responsible for `2`.
But the `VertxInstance` service is always called when the factory is started (with or without an existing Vert.x instance), that's why I said that the line `when there is no instance associated with the calling thread.` doesn't seem correct (it implies that it gets called only when there is no Vert.x instance).

Reply:
> Like .... why don't we just use the Vertx.currentContext() if there is one? Why do we insist on it being the one coming from "our" Vert.x?

Doesn't it makes sure that 1. we are running in a context (it will create a new one if one doesn't exist) and 2. an existing session run the operations always in the same context and that should imply the same thread (this is the part where I'm never sure).

Reply:
I think I need to think more about this

Reply:
> I think I need to think more about this

Right, me too.

And I think perhaps we should wait until @Sanne gets back and discuss it thoroughly.

Reply:
> Doesn't it makes sure that 1. we are running in a contex

I remember in the beginning we had a lot of issues with operations not running in the right context when using Quarkus.